packages/explat-client/src/test/create-explat-client.ts

```diff
@@ -0,0 +1,697 @@
+/**
+ * External dependencies
+ */
+// This is required to fix the "regeneratorRuntime is not defined" error
+import '@automattic/calypso-polyfills';
+
+/**
+ * Internal dependencies
+ */
+import { createExPlatClient } from '../create-explat-client';
+import {
+	delayedValue,
+	ONE_DELAY,
+	validExperimentAssignment,
+	ZERO_DELAY,
+} from '../internal/test-common';
+import * as Timing from '../internal/timing';
+import type { Config, ExperimentAssignment } from '../types';
+
+const spiedMonotonicNow = jest.spyOn( Timing, 'monotonicNow' );
+
+const mockedFetchExperimentAssignment = jest.fn();
+const mockedGetAnonId = jest.fn();
+const mockedLogError = jest.fn();
+const createMockedConfig = ( isDevelopmentMode = false ): Config => ( {
+	logError: mockedLogError,
+	fetchExperimentAssignment: mockedFetchExperimentAssignment,
+	getAnonId: mockedGetAnonId,
+	isDevelopmentMode,
+} );
+const DEVELOPMENT_MODE = true;
+
+function mockFetchExperimentAssignmentToMatchExperimentAssignment(
+	experimentAssignment: ExperimentAssignment
+) {
+	spiedMonotonicNow.mockImplementationOnce( () => experimentAssignment.retrievedTimestamp );
+	mockedFetchExperimentAssignment.mockImplementationOnce( () =>
```

Comment:
This mock function doesn't match the signature of the function we're mocking. We should accept the `experimentName` and `anonId` parameters and check their value. The formatting is a mess, but I think that something like this would work:

```typescript
function mockFetchExperimentAssignmentToMatchExperimentAssignment(
	{ experimentAssignment, expectedAnonId } :
	{ experimentAssignment: ExperimentAssignment; expectedAnonId: string | null; }
) {
	spiedMonotonicNow.mockImplementationOnce( () => experimentAssignment.retrievedTimestamp );
	mockedFetchExperimentAssignment.mockImplementationOnce( ( experimentName, anonId ) => {
		expect( experimentName ).toBe( experimentAssignment.experimentName );
		expect( anonId ).toBe( expectedAnonId );
		...
```

See also below for how it's useful in the context of specific tests.

Reply:
I have just ended up checking the `mockedConfig.requestExperimentAssignment.mock.calls`.

Reply:
Why? How is it better than using a more realistic mock?

Reply:
<img width="1355" alt="Screen Shot 2021-02-18 at 4 58 17 pm" src="https://user-images.githubusercontent.com/971886/108332226-86b23f00-720a-11eb-9fd6-17c6e38f85ca.png">

It gets a bit complicated as jest works by throwing errors and the error is occurring within a config function. I am also sometimes returning a different experimentAssignment to what was requested to check other things.

I could still add more checks on the requestExperimentAssignment calls, maybe snapshotting them?

Reply:
Ah, never mind then. I think what we have is good enough.